### DISCUSSION TITLE
name for tabular learner factory

### DISCUSSION THREAD

**Discussion by jeromedockes** [skrub-data/skrub#938] [General]
[#926](https://github.com/skrub-data/skrub/pull/926) introduces a function to easily get a learner that works on tabular data (by chaining a TableVectorizer, missing value imputation if necessary, and an estimator).

It is used like this:

```python
>>> from skrub import table_learner
>>> table_learner('regressor')
Pipeline(steps=[('tablevectorizer',
                 TableVectorizer(high_cardinality_transformer=MinHashEncoder(),
                                 low_cardinality_transformer=ToCategorical())),
                ('histgradientboostingregressor',
                 HistGradientBoostingRegressor(categorical_features='from_dtype'))])
>>> from sklearn.linear_model import Ridge
>>> table_learner(Ridge(solver='lsqr'))
Pipeline(steps=[('tablevectorizer', TableVectorizer()),
                ('simpleimputer', SimpleImputer()),
                ('ridge', Ridge(solver='lsqr'))])

```

What should be the name of this function (`make_tabular_pipeline` in the example)?

I'm not using a poll because in polls adding options removes existing votes. Please upvote the comment containing the option you like and don't hesitate to add options!

**Comment by jeromedockes** [comment]
`make_learner`

**Reply by TheooJ** [reply]
Although I like this name, I wonder whether most users will be familiar with the term "learner" (or "règle d'apprentissage", something that takes data as an input and returns a predictor, section 1.3 of https://www.imo.universite-paris-saclay.fr/~sylvain.arlot/enseign/2020-21_M2/1-fondamentaux.pdf would be my definition)

**Comment by jeromedockes** [comment]
`make_skrub_learner`

**Comment by jeromedockes** [comment]
`make_tabular_learner`

**Comment by jeromedockes** [comment]
`make_skrub_pipeline`

**Comment by jeromedockes** [comment]
`make_tabular_pipeline`

**Reply by Vincent-Maladiere** [reply]
I agree that the function should have "pipe" or "pipeline" since this is what is returned in the end. Since we're dealing with defaults for tabular data, I'm +1 for this option.

**Reply by GaelVaroquaux** [reply]
I'm actually not that convinced. The fact that it is a pipeline is a technical detail. A pipeline still has a fit and a predict. A non advanced user doesn't know what a pipeline is, and that function is specifically geared to a non advanced user.

**Comment by GaelVaroquaux** [comment]
I agree with Theo that users can probably relate "learner" more to their needs. 

make_tabular_learner is too long, in my opinion (long names impose a real toll on users).

Thinking in different directions: 
- "tabularize(model)"
- "prep_learner(model)"

The last one is fun, because "to prep oneself" can involve putting on clothes, which is what we are doing to the learner.

**Comment by jeromedockes** [comment]
`tabularize`

**Comment by jeromedockes** [comment]
`prep_learner`

**Comment by GaelVaroquaux** [comment]
`wrap_learner`

**Comment by GaelVaroquaux** [comment]
`pipe_learner`

**Reply by GaelVaroquaux** [reply]
Added this in case we want to make it explicit that it creates a pipeline (which could be a good idea)

**Reply by jeromedockes** [reply]
to me 
- this sounds like something that learns pipes :sweat_smile: 
- once I see it's a verb, I still find it a bit weird because I would say we pipe some data to the learner (as we pipe a stream to a process), not that we pipe the learner itself

**Comment by glemaitre** [comment]
What about `vectorized_learner(learner)`?

**Reply by TheooJ** [reply]
I also like `vectorize(learner)`

**Reply by glemaitre** [reply]
I think I'm also fine with the short `vectorize`

**Reply by jeromedockes** [reply]
to me vectorizing code is making it applicable to multiple data, as in [numpy.vectorize](https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html) or simd instructions. then there is vectorizing data, ie feature extraction / tablevectorizer. So I'm not strongly opposed but I think `vectorize(learner)` could be a bit confusing

**Reply by TheooJ** [reply]
I wasn't aware of this kind of vectorization
In that case doesn't this make the `TableVectorizer` confusing too ?

**Reply by jeromedockes** [reply]
not really -- vectorizing data usually means feature extraction, as done in the tablevectorizer so that's fine. but here `learner` is not data, so the meaning of `vectorize(learner)` is less clear IMO

**Reply by ogrisel** [reply]
I think "vectorize" has ambiguous meanings in ML / array computing / HPC communities. I would much prefer "tabularize".

**Comment by jeromedockes** [comment]
`skrubbify(learner)` :joy:

**Reply by glemaitre** [reply]
If we call the class `Skrubber`, then why not :)

**Comment by LeoGrin** [comment]
`make_default_pipeline`, `make_easy_pipepline`,`make_simple_pipeline`, or `make_adapted_pipeline`?

**Reply by jeromedockes** [reply]
I like `make_default_pipeline`

**Reply by GaelVaroquaux** [reply]
Too long

**Comment by GaelVaroquaux** [comment]
The name was chosen with a Twitter poll to consolidate the above discussion: https://x.com/skrub_data/status/1802687002657124557?t=RKbDh_hjE9-lxQiaC0oFcg&s=19

Name: tabular_learner
